tests/test_love_manager_client.py

```diff
@@ -0,0 +1,316 @@
+# This file is part of ts_salobj.
+#
+# Developed for the Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import json
+import asyncio
+import logging
+import unittest
+import contextlib
+import websockets
+
+
+from love.producer import LoveManagerClient, LoveManagerMessage
+from love.producer.test_utils import cancel_task
+
+
+class TestLoveManagerClient(unittest.IsolatedAsyncioTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.log = logging.getLogger(__name__)
+        cls.standard_timeout = 5.0
+        cls.pool_timeout = 0.1
+
+    def setUp(self):
+        self.love_manager_client = LoveManagerClient(self.log)
+        self.love_manager_message = LoveManagerMessage("UnitTest")
+        self.received_data = dict()
+
+        self.sample_summary_state = dict(summaryState=4)
+
+    async def asyncTearDown(self):
+        await self.love_manager_client.close()
+
+    async def test_url(self):
+
+        async with self.configure_environment_with_websocket_host():
+            self.assertIsInstance(self.love_manager_client.url, str)
+
+    def test_url_no_manager_hostname(self):
+
+        back_up_websocket_host = (
+            os.environ.pop("WEBSOCKET_HOST") if "WEBSOCKET_HOST" in os.environ else None
+        )
+
+        try:
+            with self.assertRaises(RuntimeError):
+                self.love_manager_client.url
+        finally:
+            if back_up_websocket_host is not None:
+                os.environ["WEBSOCKET_HOST"] = back_up_websocket_host
+
+    async def test_create_producers(self):
+
+        components = self.create_producers()
+
+        self.assertEqual(len(components), len(self.love_manager_client.producers))
+
+        for component, producer in zip(components, self.love_manager_client.producers):
+            self.assertEqual(component, producer.component_name)
+
+    async def test_handle_connection_with_manager_no_producers(self):
```

Comment:
How does this works?

Reply:
So, this is a very simple connectivity test. 

The idea is to make sure that, when the manager client connects with server, the `connected_task` result is `True` and `done_task` is not done, meaning, the manager client is running.

Since this is a basic connectivity unit test, there's no reason to create producers, hence "no_producers".

I added a comment in the test to explain this.

Reply:
I understand, thanks!